Inspector/macho.c

```diff
@@ -0,0 +1,378 @@
+    if (mach_hdr == NULL) return NULL;
+    
+    size_t header_size;
+    if (is_magic_64(mach_hdr->magic)) {
+        header_size = sizeof(struct mach_header_64);
+    } else {
+        header_size = sizeof(struct mach_header);
+    }
+    
+    void *start = (void *)mach_hdr + header_size;
+    return (struct load_command *)start;
+}
+
+uint32_t get_load_command_count(const struct mach_header *mach_hdr) {
+    if (mach_hdr == NULL) return 0;
+    
+    return mach_hdr->ncmds;
+}
+
+const struct load_command **get_load_commands(const struct mach_header *mach_hdr) {
+    uint32_t command_count = get_load_command_count(mach_hdr);
+    void *next_record = (void *)get_first_load_command(mach_hdr);
+
+    const struct load_command **load_commands = malloc(command_count * sizeof(struct load_command *));
+    for (int i = 0; i < command_count; i++) {
+        const struct load_command *cmd = (struct load_command *)next_record;
+        load_commands[i] = cmd;
+        next_record += cmd->cmdsize;
+    }
+
+    return load_commands;
+}
+
+uint32_t get_load_command_count_by_type(const struct mach_header *mach_hdr, uint32_t type) {
+    uint32_t count = 0;
+    
+    uint32_t command_count = get_load_command_count(mach_hdr);
+    const struct load_command **load_commands = get_load_commands(mach_hdr);
+    for (int i = 0; i < command_count; i++) {
+        const struct load_command *cmd = load_commands[i];
+        if (cmd->cmd == type) {
+            count++;
+        }
+    }
+    free(load_commands);
+    
+    return count;
+}
+
+const struct load_command **get_load_commands_by_type(const struct mach_header *mach_hdr, uint32_t type) {
+    uint32_t command_count = get_load_command_count(mach_hdr);
+    const struct load_command **load_commands = get_load_commands(mach_hdr);
+    uint32_t type_count = get_load_command_count_by_type(mach_hdr, type);
+    const struct load_command **type_commands = malloc(type_count * sizeof(struct load_command *));
+    uint32_t type_index = 0;
+    for (int i = 0; i < command_count; i++) {
+        const struct load_command *cmd = load_commands[i];
+        if (cmd->cmd == type) {
+            type_commands[type_index] = cmd;
+            type_index++;
+        }
+    }
+    free(load_commands);
+    
+    return type_commands;
+}
+
+const char *get_name(uint32_t image_no) {
+    return _dyld_get_image_name(image_no);
+}
+
+const uint8_t *get_uuid(uint32_t image_no) {
+    const struct mach_header *mach_hdr = _dyld_get_image_header(image_no);
+    uint32_t command_count = get_load_command_count_by_type(mach_hdr, LC_UUID);
+    const struct load_command **load_commands = get_load_commands_by_type(mach_hdr, LC_UUID);
+    for (int i = 0; i < command_count;) {
+        return ((const struct uuid_command *)load_commands[i])->uuid;
+    }
+    free(load_commands);
+    
+    return NULL;
+}
+
+uint8_t *get_cs_blob_by_magic(uint32_t image_no, enum cs_magic magic) {
```

Comment:
This assumes that there is no more than one blob with a given magic. That may be true, I'm not sure.